packages/openapi3/test/models.test.ts

```diff
@@ -685,6 +685,25 @@ describe("openapi3: models", () => {
     });
   });
 
+  it("defines oneOf schema for property of a union with @oneOf decorator", async () => {
+    const openApi = await openApiFor(`
+        model Foo {
+          @oneOf
+          bar: string | int32;
```

Comment:
curious if @oneOf should still work in cases like 
```
@oneOf
bar: string | string;
```
See https://github.com/microsoft/typespec/issues/2345